server/routes/SolutionRoutes.js

```diff
@@ -32,18 +37,26 @@ router.get("/:id", Solution.get);
 // Sorting (by date or upvote) happens on the client side.
 // router.get("/:id", Challenge.getComments);
 
-// Create a new solution in the database.
-router.post("/", upload.single("diagram"), Solution.create);
 
-// Edit a solution in the database.
-router.put("/", upload.single("diagram"), Solution.edit);
+// Create a new solution in the database, allowing file upload and restricting access to certain roles.
+router.post("/", checkRole(['user']), upload.single("diagram"), Solution.create);
```

Comment:
is admin not allowed to POST diagrams?

Reply:
Will update that. I will be creating a new branch for the check role stuff as I need to figure out session cookies before working on the roles